test/integration/controlplane/transformation/kmsv2_transformation_test.go

```diff
@@ -201,6 +204,44 @@ resources:
 	}
 	defer test.cleanUp()
 
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// the global metrics registry persists across test runs - reset it here so we can make assertions
+	copyConfig := rest.CopyConfig(test.kubeAPIServer.ClientConfig)
+	copyConfig.GroupVersion = &schema.GroupVersion{}
+	copyConfig.NegotiatedSerializer = unstructuredscheme.NewUnstructuredNegotiatedSerializer()
+	rc, err := rest.RESTClientFor(copyConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := rc.Delete().AbsPath("/metrics").Do(ctx).Error(); err != nil {
+		t.Fatal(err)
+	}
+
+	// assert that the metrics we collect during the test run match expectations
+	wantMetricStrings := []string{
```

Comment:
should this return all metrics that do not contain `_seconds`? if so, how about `invalid_key_id_from_status_total`

Reply:
Metrics are lazily initialized on the first time they are set.  Since `invalid_key_id_from_status_total` only gets used in a specific error case, it will not be present in this test.